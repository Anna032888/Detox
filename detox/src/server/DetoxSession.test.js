describe('DetoxSession', () => {
  /**
   * @type {typeof require('./DetoxSession')}
   */
  let DetoxSession;
  /**
   * @type {typeof require('./DetoxConnection')}
   */
  let DetoxConnection;
  /**
   * @type {DetoxSession}
   */
  let session;
  let logger;

  beforeEach(() => {
    jest.mock('../utils/logger');
    logger = require('../utils/logger');

    DetoxConnection = jest.genMockFromModule('./DetoxConnection');
    DetoxSession = require('./DetoxSession');
    session = new DetoxSession('aSession');
  });

  describe('edge cases', () => {
    it('should warn on .disconnect(null)', () => {
      session.disconnect(null);
      expect(getWarning()).toMatchSnapshot();
    });

    it('should warn on .disconnect(<unknown connection>)', () => {
      session.disconnect(new DetoxConnection());
      expect(getWarning()).toMatchSnapshot();
    });

    it('should warn on non-empty app assignment', () => {
      session.app = new DetoxConnection();
      session.app = new DetoxConnection();
      expect(getWarning()).toMatchSnapshot();
    });

    it('should warn on non-empty tester assignment', () => {
      session.tester = new DetoxConnection();
      session.tester = new DetoxConnection();
      expect(getWarning()).toMatchSnapshot();
    });
  });

  function getWarning(index = 0) {
    return logger.error.mock.calls[index][0];
  }
});
