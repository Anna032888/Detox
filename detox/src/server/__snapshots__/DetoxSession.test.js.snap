// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetoxSession edge cases should warn on .disconnect(<unknown connection>) 1`] = `
"DetoxInvariantError: cannot disconnect an unknown connection from the session aSession
Please report this issue on our GitHub tracker:
https://github.com/wix/Detox/issues"
`;

exports[`DetoxSession edge cases should warn on .disconnect(null) 1`] = `
"DetoxInvariantError: DetoxSession(aSession).prototype.disconnect(connection) expects a non-null argument.
Please report this issue on our GitHub tracker:
https://github.com/wix/Detox/issues"
`;

exports[`DetoxSession edge cases should warn on non-empty app assignment 1`] = `
"DetoxInvariantError: the app is already connected to the session aSession
Please report this issue on our GitHub tracker:
https://github.com/wix/Detox/issues"
`;

exports[`DetoxSession edge cases should warn on non-empty tester assignment 1`] = `
"DetoxInvariantError: the tester is already connected to the session aSession
Please report this issue on our GitHub tracker:
https://github.com/wix/Detox/issues"
`;
